bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,761 @@
+
+    private @Nullable DaikinMadokaConfiguration config;
+
+    private @Nullable ExecutorService commandExecutor;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    // UART Processor is in charge of reassembling chunks
+    private BRC1HUartProcessor uartProcessor = new BRC1HUartProcessor(this);
+
+    private volatile @Nullable BRC1HCommand currentCommand = null;
+
+    private MadokaSettings madokaSettings = new MadokaSettings();
+
+    public DaikinMadokaHandler(Thing thing) {
+        super(thing);
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        logger.debug("[{}] Start initializing!", super.thing.getUID().getId());
+
+        // Load Configuration
+        config = getConfigAs(DaikinMadokaConfiguration.class);
+
+        logger.debug("[{}] Parameter value [refreshInterval]: {}", super.thing.getUID().getId(),
+                config.refreshInterval);
+        logger.debug("[{}] Parameter value [commandTimeout]: {}", super.thing.getUID().getId(), config.commandTimeout);
+
+        if (getBridge() == null) {
+            logger.debug("[{}] Bridge is null. Exiting.", super.thing.getUID().getId());
+            return;
+        }
+
+        this.commandExecutor = Executors
+                .newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+
+        this.refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            // It is useless to refresh version all the time ! Just once.
+            if (this.madokaSettings.getCommunicationControllerVersion() == null
+                    || this.madokaSettings.getRemoteControllerVersion() == null) {
+                submitCommand(new GetVersionCommand());
+            }
+            submitCommand(new GetIndoorOutoorTemperatures());
+            submitCommand(new GetOperationmodeCommand());
+            submitCommand(new GetPowerstateCommand()); // always keep the "GetPowerState" aftern the "GetOperationMode"
+            submitCommand(new GetSetpointCommand());
+            submitCommand(new GetFanspeedCommand());
+        }, 10, this.config.refreshInterval, TimeUnit.SECONDS);
+
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("[{}] dispose()", super.thing.getUID().getId());
+
+        dispose(refreshJob);
+        dispose(commandExecutor);
+        dispose(currentCommand);
+
+        // Unsubscribe to characteristic notifications
+        if (this.device != null) {
+            BluetoothCharacteristic charNotif = this.device
+                    .getCharacteristic(DaikinMadokaBindingConstants.CHAR_NOTIF_UUID);
+
+            if (charNotif != null) {
+                @NonNull
+                BluetoothCharacteristic c = charNotif;
+                this.device.disableNotifications(c);
+            }
+        }
+
+        // Disconnect from the device
+        if (this.device != null) {
+            this.device.disconnect();
+        }
```

Comment:
Device disconnect/cleanup is already handled for you as part of `super.dispose()`. So you don't need to handle it yourself.